exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/auth/Authenticator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.auth;
+
+import io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder;
+import io.opentelemetry.exporter.internal.okhttp.OkHttpExporterBuilder;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/** Internal class to allow users of OTLP-OkHttp exporters to add support for authentication. */
```

Comment:
Our standard javadoc for internal classes is:
```
/**
 * This class is internal and is hence not for public use. Its APIs are unstable and can change at
 * any time.
 */
```
Can add an extra info if you want.